### PR TITLE
heartbeat on view messages

### PR DESCRIPTION
**Pull request by HagarMeir**
Signed-off-by: Hagar Meir <hagar.meir@ibm.com>

### CODE REVIEW

**Comment by HagarMeir** [comment]
https://github.com/SmartBFT-Go/consensus/issues/241

File: pkg/consensus/consensus.go [hunk 1]
```diff
@@ -209,7 +208,7 @@ func (c *Consensus) BroadcastConsensus(m *protos.Message) {
 func (c *Consensus) proposalMaker() *algorithm.ProposalMaker {
 	return &algorithm.ProposalMaker{
 		State:           c.state,
-		Comm:            c,
+		Comm:            c.controller,
```
Comment (yacovm) [inline pkg/consensus/consensus.go:205]: The BroadcastConsensus method can now be removed.

File: internal/bft/controller.go [hunk 2]
```diff
@@ -280,6 +284,29 @@ func (c *Controller) respondToStateTransferRequest(sender uint64) {
 	c.Comm.SendConsensus(sender, msg)
 }
 
+func (c *Controller) convertViewMessageToHeartbeat(m *protos.Message) *protos.Message {
+	var view, seq uint64
+	switch m.GetContent().(type) {
+	case *protos.Message_PrePrepare:
```
Comment (yacovm) [inline internal/bft/controller.go]: we have methods that extract the seq and view number in util.go, you can use them.

**Review by yacovm — CHANGES_REQUESTED** [review]
a few nits + that thing i sent via slack

**Pull request closed without merging**